datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -333,21 +333,87 @@ pub fn date_trunc(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     })
 }
 
-fn date_bin_single(stride: i64, source: i64, origin: i64) -> i64 {
+trait DateBinInterval {
+    fn date_bin_interval(&self, stride: i64, source: i64, origin: i64) -> i64;
+}
```

Comment:
I create this trait for it to work with an available closure because I now need 2 different ways to handle date_bin  intervals nanoseconds and months

Reply:
This trait is no longer needed from the comment below